### ISSUE TITLE
Chaining Identity Server with B2C

### ISSUE THREAD

**Issue by JTAtkins** [DuendeSoftware/Support#887]
This is a general and possibly stupid question, but trying to think out of the box, and I think people get into trouble trying to write their own when it comes to identity.  

Azure AD B2C doesnt support Device Code, but does allow custom IDP.  I'm wondering if anyone has tried to chain Identity server into other IdPs for a particular flow and extend them securely?  

In this case the app is heavily tied to B2C/MSAL, but needs device code now.   In this case I'm thinking you would have Identity Server providing the device code endpoint client -> /devicecode, client->login (identity server - Identity Server delegates to b2c for login challenge, user authenticates, etc....  

Basically Identity Server would be a proven secure middleman to extend b2c.  There are various proxies/solutions to this problem around GitHub, but it really comes down to dont write your own security if you can help it.  It is very hard to not introduce issues in my opinon.  

Any thoughts are greatly appreciated.

Thanks!

**Comment by AndersAbel** [comment]
Yes, the solution you outline would be possible.

Please note however that the tokens issued by IdentityServer as part of the device flow will have the IdentityServer as the issuer. The application and any APIs it is calling needs to be updated to trust IdentityServer as the token issuer.

If that is not possible (you mention that the application is tightly bound to B2C/MSAL) it is possible to make customizations to IdentityServer to make the original B2C access token available to the client application.

**Comment by AndersAbel** [comment]
Do you have any more questions on this or can we go ahead and close the issue?
